### PR TITLE
Feature - Algorithm Management

### PR DESCRIPTION
**Pull request by friedow**
This solves #44.

### CODE REVIEW

File: src/utils/Utils.ts [hunk 1]
```diff
@@ -86,6 +87,50 @@ export default class Utils {
...
+    return [
+      {
+        id: 'name',
+        firstValue: 'Name',
+        secondValue: algorithm.name,
+      },
+      {
+        id: 'inputs',
+        firstValue: 'Inputs',
+        secondValue: algorithm.inputs.join(', '),
```
Comment (Munin33) [inline src/utils/Utils.ts]: please display the name instead of the id of the resource type here.

File: src/utils/Utils.ts [hunk 2]
```diff
@@ -86,6 +87,50 @@ export default class Utils {
...
+      },
+      {
+        id: 'outpus',
+        firstValue: 'Output',
+        secondValue: algorithm.outputs,
```
Comment (Munin33) [inline src/utils/Utils.ts]: please display the name instead of the id of the resource type here.

File: src/utils/Utils.ts [hunk 3]
```diff
@@ -86,6 +87,50 @@ export default class Utils {
     });
   }
 
+  public static optimizationAlgorithmsToList(algorithms: OptimizationAlgorithm[], onClick?: clickHandler) {
+    return algorithms.map((algorithm) => {
+      return {
+        id: algorithm.id || algorithm.name,
+        firstValue: algorithm.name,
+        secondValue:
+          `( ${algorithm.inputs.join(', ')} ) => ${algorithm.outputs}`,
```
Comment (Munin33) [inline src/utils/Utils.ts]: please display the names instead of ids

File: src/utils/Utils.ts [hunk 4]
```diff
@@ -86,6 +87,50 @@ export default class Utils {
...
+        thirdValue: '',
+        link: onClick ? {
+          onClick: () => { onClick(algorithm.id || algorithm.name); },
+        } : {
+          link: { name: 'algorithm', params: { id: algorithm.id }},
```
Comment (Munin33) [inline src/utils/Utils.ts:100]: why do we need a link here? we are already on this page.
I know, the resource type view has also the link to itself, but i would ask the same question there.

File: src/views/CreateAlgorithm.vue [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    <ViewHeader
+      title="What information does the algorithm need?"
+      :backLink="{ onClick: previousStep }"
+    />
+    <ListSection title="Algorithm Inputs">
+      <li class="options slot input" v-for="resourceType in selectableResourceTypes" :key="resourceType.id" @click="toggleInputSelection(resourceType)">
+        <div class="checkbox" :class="{checked: resourceType.selected}" />
+        <div>
+          <p class="first-value">{{resourceType.name}}</p>
+          <p class="second-value">{{resourceType.parentType ? resourceType.parentType.name : ''}}</p>
```
Comment (Munin33) [inline src/views/CreateAlgorithm.vue]: I would add the string "Parent type: " before the actual type to make clear why this name is there.

**Review by Munin33 — CHANGES_REQUESTED** [review]

File: src/utils/Utils.ts (hunk 4 above)
Comment (friedow) [reply src/utils/Utils.ts:100]: The thing is, this util function builds the single algorithm view as well as the list.

File: src/views/CreateAlgorithm.vue (hunk 5 above)
Comment (friedow) [reply src/views/CreateAlgorithm.vue]: Done

**Review by Munin33 — APPROVED** [review]

**Pull request merged**